parser/conceptParser.go

```diff
@@ -232,14 +232,34 @@ func CreateConceptsDictionary() (*gauge.ConceptDictionary, *ParseResult) {
 		res.ParseErrors = append(res.ParseErrors, errs...)
 		res.Ok = false
 	}
-	vRes := validateConcepts(conceptsDictionary)
-	if len(vRes.CriticalErrors) > 0 {
+	vRes := ValidateConcepts(conceptsDictionary)
+	if len(vRes.ParseErrors) > 0 {
 		res.Ok = false
-		res.CriticalErrors = append(res.CriticalErrors, vRes.CriticalErrors...)
+		res.ParseErrors = append(res.ParseErrors, vRes.ParseErrors...)
 	}
 	return conceptsDictionary, res
 }
 
+func AddConcept(concepts []*gauge.Step, file string, conceptDictionary *gauge.ConceptDictionary) []ParseError {
```

Comment:
code seems to be duplicated in AddConcepts function. AddConcepts function could call AddConcept function.